hmp/models/fixedn.py

```diff
@@ -293,58 +296,65 @@ def fit(
             resetwarnings()
 
         self.lkhs = np.array([x[0] for x in estimates])
-        self.mags = np.array([x[1] for x in estimates])
-        self.pars = np.array([x[2] for x in estimates])
-        self.traces = np.array([x[3] for x in estimates])
-        self.param_dev = np.array([x[4] for x in estimates])
-        if self.starting_points > 1 and self.return_max:
+        if self.starting_points > 1 :
             max_lkhs = np.argmax(self.lkhs)
-            self.lkhs = self.lkhs[[max_lkhs]]
-            self.mags = self.mags[[max_lkhs]]
-            self.pars = self.pars[[max_lkhs]]
-            self.traces = self.traces[[max_lkhs]]
-            self.param_dev = self.param_dev[[max_lkhs]]
-
+        else:
+            max_lkhs = 0
+        self.lkhs = self.lkhs[max_lkhs]
+        self.magnitudes =  np.array([x[1] for x in estimates])[max_lkhs]
+        self.parameters = np.array([x[2] for x in estimates])[max_lkhs]
+        self.traces = np.array([x[3] for x in estimates])[max_lkhs]
+        self.param_dev = np.array([x[4] for x in estimates])[max_lkhs]
+        
+        self.level_dict = level_dict
         self.levels = levels
         self.n_levels = n_levels
         self.mags_map = mags_map
         self.pars_map = pars_map
 
         self._fitted = True
 
-    def transform(self, trial_data, level_id=0):
+    def transform(self, trial_data):
+        n_levels, levels, clabels = self._level_constructor(
+                trial_data, self.level_dict
+            )
         all_event_probs = []
         all_likelihoods = []
-        for i_sp in range(self.mags.shape[0]):
-            likelihood, eventprobs = self.estim_probs(
-                trial_data,
-                self.mags[i_sp][level_id],
-                self.pars[i_sp][level_id],
-                np.zeros((self.n_events+1)).astype(int),
-                lkh_only=False
-            )
-            part = trial_data.coords["participant"].values
-            trial = trial_data.coords["trials"].values
+        for c in range(n_levels):
+            locations = np.zeros((n_levels, self.n_events + 1,), dtype=int)
+            magnitudes_level = self.magnitudes[
+                c, self.mags_map[c, :] >= 0, :
+            ]  # select existing magnitudes
+            parameters_level = self.parameters[c, self.pars_map[c, :] >= 0, :]  # select existing params
+            likelihood, eventprobs = \
+                self.estim_probs(
```

Comment:
```suggestion
            likelihood, eventprobs =  self.estim_probs(
```
Generally suggested is not to use line breaks if not absolutely necessary.